Modules/linAnalyzer.py

```diff
@@ -196,308 +106,303 @@ def hash_calculator(self, filename, report_object):
-        linrep["number_of_sections"] = len(self.binary.sections)
-        linrep["number_of_segments"] = len(self.binary.segments)
-        self.hash_calculator(self.target_file, linrep)
-        self.SecCheck()
-
-    # Gathering sections
-    def SectionParser(self):
-        secTable = Table(title="* Informations About Sections *", title_justify="center", title_style="bold italic cyan")
-        secTable.add_column("[bold green]Section Names", justify="center")
-        secTable.add_column("[bold green]Size(bytes)", justify="center")
-        secTable.add_column("[bold green]Offset", justify="center")
-        secTable.add_column("[bold green]Virtual Address", justify="center")
-        secTable.add_column("[bold green]Entropy", justify="center")
+        self.report["machine_type"] = self.binary.header.machine_type.name
+        self.report["binary_entrypoint"] = str(hex(self.binary.entrypoint))
+        self.report["number_of_sections"] = len(self.binary.sections)
+        self.report["number_of_segments"] = len(self.binary.segments)
+        self.calc_hashes(self.target_file)
+        self.check_bin_security()
+
+    def parse_sections(self):
+        section_t = init_table(
+            "Section Names", "Size(bytes)",
+            "Offset", "Virtual Address", "Entropy",
+            col_prefix="[bold green]",
+            title="* Informations About Sections *",
+        )
         for sec in self.binary.sections:
-            if sec.name != "" and sec.name != " ":
-                secTable.add_row(
-                    f"[bold red]{sec.name}", 
-                    str(sec.size),
-                    str(hex(sec.offset)),
-                    str(hex(sec.virtual_address)),
-                    str(sec.entropy)
-                )
-                linrep["sections"].append(
-                    {
-                        "name": sec.name,
-                        "size": sec.size,
-                        "offset": str(hex(sec.offset)),
-                        "virtual_address": str(hex(sec.virtual_address)),
-                        "entropy": str(sec.entropy)
-                    }
-                )
-        print(secTable)
-
-    # Gathering segments
-    def SegmentParser(self):
-        segTable = Table(title="* Informations About Segments *", title_justify="center", title_style="bold italic cyan")
-        segTable.add_column("[bold green]Segments", justify="center")
-        segTable.add_column("[bold green]Contained Sections", justify="center")
+            if sec.name.strip() == "":
+                continue
+            metadata = { # TODO consider OrderedDict type instead
+                "name": sec.name, "size": str(sec.size),
+                "offset": str(hex(sec.offset)),
+                "virtual_address": str(hex(sec.virtual_address)),
+                "entropy": str(sec.entropy)
+            }
+            self.report["sections"].append(metadata)
+            metadata["name"] = "[bold red]"+metadata["name"]

```

Comment:
@CYB3RMX

ok, on my lunch break right now, just a quick update, I apologize for the delay:
- just reread the changes and I think I just spotted the bug regarding the rich markup ending up in the report file

It's caused by me modifying `metadata` after having appended it (in it's valid form) to the report. Because this is a reference though by default, not a copy, the modification on the local variable also applies to the report. It's a relatively easy fix using `copy.deepcopy` instead.

Before now, I sadly didn't get around to taking care of any of the further steps including not getting started on the test system setup.

But tonight/evening, I'll quickly setup a debian box 💻 and verify the fix, which was caused by me being unaware of the reference pass, thus subsequently modifying the report unintentionally [here](https://github.com/CYB3RMX/Qu1cksc0pe/pull/58/files#diff-a9181e8c173e93a514cc6542354ce910520c172bfcc1c9698837b0204d1da2e5R182)

after that verification, we can hopefully finally get this shipped (: